packages/data/src/components/use-select/index.js

```diff
@@ -1,3 +1,4 @@
+'use no memo';
```

Comment:
This disables memoization of the conditional call in `useSelect`:
```js
return staticSelectMode : _useStaticSelect() : _useMappingSelect();
```
The React Compiler things that the `_use` functions are regular function calls that it can memoize. But it can't, because that leads to the inner hook being called only sometimes. Breaking the rules of hooks in a breaking way.

Reply:
Makes sense to me! I think this is a good compromise to allow moving this work forward.